pkg/operator/target_config_reconciler.go

```diff
@@ -0,0 +1,627 @@
+}
+
+func (c *TargetConfigReconciler) manageServiceAccount(leaderWorkerSetOperator *leaderworkersetapiv1.LeaderWorkerSetOperator) (*v1.ServiceAccount, bool, error) {
+	required := resourceread.ReadServiceAccountV1OrDie(bindata.MustAsset("assets/lws-operator/serviceaccount.yaml"))
+	required.Namespace = c.namespace
+	ownerReference := metav1.OwnerReference{
+		APIVersion: "operator.openshift.io/v1",
+		Kind:       "LeaderWorkerSetOperator",
+		Name:       leaderWorkerSetOperator.Name,
+		UID:        leaderWorkerSetOperator.UID,
+	}
+	required.OwnerReferences = []metav1.OwnerReference{
+		ownerReference,
+	}
+	controller.EnsureOwnerRef(required, ownerReference)
+
+	return resourceapply.ApplyServiceAccount(c.ctx, c.kubeClient.CoreV1(), c.eventRecorder, required)
+}
+
+func (c *TargetConfigReconciler) manageCustomResourceDefinition(leaderWorkerSetOperator *leaderworkersetapiv1.LeaderWorkerSetOperator) (*apiextensionv1.CustomResourceDefinition, bool, error) {
+	required := resourceread.ReadCustomResourceDefinitionV1OrDie(bindata.MustAsset("assets/lws-operator/leaderworkerset.x-k8s.io_leaderworkersets.yaml"))
+	ownerReference := metav1.OwnerReference{
+		APIVersion: "operator.openshift.io/v1",
+		Kind:       "LeaderWorkerSetOperator",
+		Name:       leaderWorkerSetOperator.Name,
+		UID:        leaderWorkerSetOperator.UID,
+	}
+	required.OwnerReferences = []metav1.OwnerReference{
+		ownerReference,
+	}
+	controller.EnsureOwnerRef(required, ownerReference)
+
+	if required.Spec.Conversion != nil &&
+		required.Spec.Conversion.Webhook != nil &&
+		required.Spec.Conversion.Webhook.ClientConfig != nil &&
+		required.Spec.Conversion.Webhook.ClientConfig.Service != nil {
+		required.Spec.Conversion.Webhook.ClientConfig.Service.Namespace = c.namespace
+	}
+
+	return resourceapply.ApplyCustomResourceDefinitionV1(c.ctx, c.apiextensionClient.ApiextensionsV1(), c.eventRecorder, required)
+}
+
+func (c *TargetConfigReconciler) manageMutatingWebhook(leaderWorkerSetOperator *leaderworkersetapiv1.LeaderWorkerSetOperator) (*admissionv1.MutatingWebhookConfiguration, bool, error) {
+	required := resourceread.ReadMutatingWebhookConfigurationV1OrDie(bindata.MustAsset("assets/lws-operator/mutatingwebhook.yaml"))
+	ownerReference := metav1.OwnerReference{
+		APIVersion: "operator.openshift.io/v1",
+		Kind:       "LeaderWorkerSetOperator",
+		Name:       leaderWorkerSetOperator.Name,
+		UID:        leaderWorkerSetOperator.UID,
+	}
+	required.OwnerReferences = []metav1.OwnerReference{
+		ownerReference,
+	}
+	controller.EnsureOwnerRef(required, ownerReference)
+
+	for i, _ := range required.Webhooks {
+		if required.Webhooks[i].ClientConfig.Service != nil {
+			required.Webhooks[i].ClientConfig.Service.Namespace = c.namespace
+		}
+	}
+
+	return resourceapply.ApplyMutatingWebhookConfigurationImproved(c.ctx, c.kubeClient.AdmissionregistrationV1(), c.eventRecorder, required, resourceapply.NewResourceCache())
```

Comment:
creating a new cache everytime will cause sync event flood

Reply:
Not true about the event flood. I had a different bug in my code ^^. Either way, reusing the cache shouldn't hurt.

Reply:
Agree. Changed to use single resource cache.